frontend/src/__tests__/cypress/cypress/tests/e2e/settings/servingRuntimes/testMultiModelServingRuntimeCreation.cy.ts

```diff
@@ -0,0 +1,88 @@
+import { servingRuntimes } from '~/__tests__/cypress/cypress/pages/servingRuntimes';
+import { HTPASSWD_CLUSTER_ADMIN_USER } from '~/__tests__/cypress/cypress/utils/e2eUsers';
+import { getMultiModelPath } from '~/__tests__/cypress/cypress/utils/fileImportUtils';
+import { getMultiModelServingRuntimeInfo } from '~/__tests__/cypress/cypress/utils/fileParserUtil';
+
+let modelServingName: string;
+let metadataDisplayName: string;
+
+before(() => {
+  // TODO: Investigate and resolve 'window is not defined' error during page transition seen in ODH related to application performance
+  // Temporary workaround: Catching and ignoring this specific error to prevent test failure
+  Cypress.on('uncaught:exception', (err) => {
+    // Check if the error is about 'window is not defined'
+    if (err.message.includes('window is not defined')) {
+      // Prevent the error from failing the test
+      return false;
+    }
+    // For other errors, let them fail the test
+    return true;
+  });
+  // Load multi-model serving runtime info before tests run
+  getMultiModelServingRuntimeInfo().then((info) => {
+    modelServingName = info.multiModelServingName;
+    metadataDisplayName = info.displayName;
+    cy.log(`Loaded Multi-Model Name: ${modelServingName}`);
+    cy.log(`Loaded Multi-Model Metadata Name: ${metadataDisplayName}`);
+  });
+});
+
+describe('Verify Admins Can Import and Delete a Custom Multi-Model Serving Runtime Template By Uploading A YAML file', () => {
+  it('Admin should access serving runtimes, import a yaml file and then delete', () => {
+    // Authentication and navigation
+    cy.step('Log into the application');
+    cy.visitWithLogin('/', HTPASSWD_CLUSTER_ADMIN_USER);
+
+    // Navigate to serving runtimes and import a multi-model serving yaml
+    cy.step('Navigate to Serving Runtimes and add a new serving runtime').then(() => {
+      // TODO: Remove extended timeout once '/servingruntimes' performance is optimized
+      // Current workaround for ODH page loading performance issues
+      return cy.wrap(servingRuntimes.navigate(), { timeout: 100000 });
+    });
+
+    cy.log('Navigation completed, looking for Add button');
+    cy.then(() => {
```

Comment:
What is the purpose of using `cy.then(() ...)` for `servingRuntimes.findAddButton()` ? 

I saw there were 2 other references of `findAddButton()` that did not need this

Reply:
I was facing some performance issues when executing the tests on ODH vs. RHOAI. Initially I had to chain the commands in this way for the tests to run successfully. I've reviewed this again this morning and I've got it working with the cy.then, I'll commit that now. 

Thanks for all your comments, really appreciate it.